src/python/dxpy/utils/executable_unbuilder.py

```diff
@@ -193,7 +226,30 @@ def untar_strip_leading_slash(tarfname, path):
 
                 untar_strip_leading_slash(fname, "resources")
                 os.unlink(fname)
-                deps_to_remove.append(dep)
+                # add dep name to deps_downloaded set
+                deps_downloaded.add(dep.get("name"))
+                
+            except DXError:
+                print("Download failed on dependent file with ID {} from region {}. Will try the next permitted region.\n".format(handler_id, region),
+                        file=sys.stderr)
+                # clean up deps already downloaded
+                deps_downloaded.clear()
+                created_resources_directory = False
+                shutil.rmtree("resources")
+                break
+            
+        # if all deps have been checked without an error, mark downloading as completed
+        else:
+            download_completed = True                    
+    
+    # Check if downloading is completed in one of the enabled regions
+    # if so, files in deps_downloaded will not shown in dxapp.json
+    # if not, deps_downloaded is an empty set. So ID of all deps will be in dxapp.json
+    if not download_completed:
+        print("Downloading resources failed in all enabled regions. Please try downloading with their IDs in dxapp.json.")
```

Comment:
How about adding: "You can also skip downloading resources entirely by using the --omit-resources flag."

Reply:
Added.